iam_role.go

```diff
@@ -249,6 +249,65 @@ func (c *Client) CreateIamTrustRole(options *CreateIamRoleOptions) (*IamRoleResp
 	return cr, nil
 }
 
+func jsonDecode(request *IamRoleRequest, c *Client, operation string) (obj []byte, err error) {
+	obj, e := json.Marshal(struct {
+		IamRoleRequest
+		AccountDetails
+	}{*request, c.AccountDetails})
+
+	if e != nil {
+		err = fmt.Errorf("Error encoding IAM %s JSON: %s", operation, e)
+	}
+	return
+}
```

Comment:
Why is this method called `jsonDecode` if you are in fact encoding (or rather marshaling) the request.

Reply:
This was done in haste, and in an effort to get the conversation recorded here. This is in fact a marsheller and not a decoder